tests/unit/remote/test_slow_link_detection.py

```diff
@@ -0,0 +1,108 @@
+import logging
+
+import pytest
+from dvc.config import Config
+from dvc.remote.local.slow_link_detection import (
+    slow_link_guard,
+    SlowLinkDetectorDecorator,
+)
+from mock import Mock, patch
+
+from dvc.utils import boxify
+
+
+@patch(
+    "dvc.remote.local.slow_link_detection.SlowLinkDetectorDecorator",
+    autospec=True,
+)
+class TestSlowLinkGuard(object):
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        self.method = Mock()
+        self.remote_local_mock = Mock()
+        self.config_mock = Mock()
+        self.remote_local_mock.repo.config.config.get.return_value = (
+            self.config_mock
+        )
+
+    def _cache_config(self, slow_link_warning, cache_type):
+        def config_side_effect(section_name, _):
+            if section_name == Config.SECTION_CACHE_SLOW_LINK_WARNING:
+                return slow_link_warning
+            elif section_name == Config.SECTION_CACHE_TYPE:
+                return cache_type
+
+        self.config_mock.get.side_effect = config_side_effect
+
+    def test_should_decorate_on_slow_link_warning_and_no_cache_type(
+        self, SlowLinkDetectorDecoratorClassMock
+    ):
+        self._cache_config(slow_link_warning=True, cache_type=None)
+
+        slow_link_guard(self.method)(self.remote_local_mock)
+
+        assert 1 == SlowLinkDetectorDecoratorClassMock.call_count
+        assert 0 == self.method.call_count
+
+    @pytest.mark.parametrize(
+        "slow_link_warning, cache_type",
+        [(True, "any_cache_type"), (False, None)],
+    )
+    def test_should_not_decorate(
+        self, SlowLinkDetectorDecoratorClassMock, slow_link_warning, cache_type
+    ):
+        self._cache_config(
+            slow_link_warning=slow_link_warning, cache_type=cache_type
+        )
+
+        slow_link_guard(self.method)(self.remote_local_mock)
+
+        assert 0 == SlowLinkDetectorDecoratorClassMock.call_count
+        assert 1 == self.method.call_count
+
+
+class TestSlowLinkDetectorDecorator(object):
+    @pytest.fixture(autouse=True)
+    def setUp(self):
+        self.method_result_mock = Mock()
+        self.method_mock = Mock(return_value=self.method_result_mock)
+        self.remote_local_mock = Mock()
```

Comment:
So is this needed now?

Reply:
Do you even need a mock for it though? You are passing it as a function that your decorator should wrap and call, but I don't see you doing anything with it later.

Reply:
I guess it could be left to assure that args are passed, but naming surely is wrong now